### PR TITLE
Title hud refactor

### PR DESCRIPTION
**Pull request by kevinthegreat1**
Fix some bugs with dragging, hitbox, scaling, etc.
Cleaned up code.

Depends on #816.

https://github.com/user-attachments/assets/f476984f-0879-4eff-a52f-2f9617d4adb1

### CODE REVIEW

**Comment by kevinthegreat1** [comment]
> [!note]
> Squash please.

**Comment by kevinthegreat1** [comment]
Ready for review.

File: src/main/java/de/hysky/skyblocker/skyblock/tabhud/widget/EmptyWidget.java
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/tabhud/widget/EmptyWidget.java:1]: You probably meant to put this in the same package as the `AbstractWidget`.

File: src/main/java/de/hysky/skyblocker/utils/render/title/TitleContainerConfigScreen.java [hunk 1]
```diff
@@ -1,191 +1,152 @@
...
+		}
+		return super.keyPressed(keyCode, scanCode, modifiers);
+	}
+
+	private <T extends Enum<T>> T cycle(T current) {
+		T[] values = current.getDeclaringClass().getEnumConstants();
+		return values[(current.ordinal() + 1) % values.length];
+	}
+
+	private <T extends Enum<T>> T cycleBackwards(T current) {
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/utils/render/title/TitleContainerConfigScreen.java:118]: Can't these enum methods be static? Maybe they could go somewhere too

**Review by AzureAaron — CHANGES_REQUESTED** [review]

**Pull request merged**